src/database.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * Promise database implementation, with stricter typing.
+ * By Mia
+ * @author mia-pi-git
+ */
+import * as mysql from 'mysql2';
+import SQL, {SQLStatement} from 'sql-template-strings';
+import {Config} from './config-loader';
+
+export type SQLInput = string | number | null;
+export interface ResultRow {[k: string]: SQLInput}
+
+export const databases: PSDatabase[] = [];
+
+export class PSDatabase {
+	pool: mysql.Pool;
+	prefix: string;
+	constructor(config: {[k: string]: any} = Config.mysql) {
```

Comment:
Pass the configuration file as an argument when you instantiate the class at the top-level.

Reply:
Resolved with https://github.com/smogon/pokemon-showdown-loginserver/commit/f02e16f1f69cfb6de2b30f7dba325f4cbd55f1ec.